src/sentry/static/sentry/app/views/performance/trends/chart.tsx

```diff
@@ -55,6 +61,38 @@ function transformEventStats(data: EventsStatsData, seriesName?: string): Series
   ];
 }
 
+function transformEventStatsSmoothed(data: Series[], seriesName?: string): Series[] {
+  const currentData = data[0].data;
+  const resultData = [] as SeriesDataUnit[];
+  const firstValue = movingAverage(currentData, AVERAGE_WINDOW / 2, AVERAGE_WINDOW);
+  const lastValue = movingAverage(
+    currentData,
+    currentData.length - AVERAGE_WINDOW / 2,
+    AVERAGE_WINDOW
+  );
+
+  for (let i = 0; i < currentData.length; i++) {
+    let value = 0;
+    if (i < AVERAGE_WINDOW / 2) {
+      value = firstValue;
+    } else if (i > currentData.length - AVERAGE_WINDOW / 2) {
+      value = lastValue;
+    } else {
+      value = movingAverage(currentData, i, AVERAGE_WINDOW);
```

Comment:
I don't think this is how moving averages work. The curve for the moving average shouldn't be drawn until the 10th data point (`AVERAGE_WINDOW`) from the left.

Reply:
That is accurate, but I think it makes for a worse user experience since there's a gap at the beginning of every graph now:

![image](https://user-images.githubusercontent.com/4205004/95890458-d14e1700-0d51-11eb-9b50-4198c2d7a25b.png)

Reply:
Is there a possibility to do an extra fetch to paint these gaps? 😬

Reply:
🤔 there is, but i think that would add a lot of complexity, we'd have to update _only_ the event-stats call for trends to start an extra 20 x interval earlier (eg. 10 hours more for a 14d view). I'd rather have the gap, or use the first value for the gap.

Reply:
Could the gap at the beginning use partial data? It isn't ideal as the 'average' will line up with actual data too closely, but it might be better than a gap.

Reply:
:+1: Partial data looks much better than repeating the first value:
![image](https://user-images.githubusercontent.com/4205004/95909345-50504900-0d6c-11eb-83c4-d59bbf5a77a0.png)